### PR TITLE
Add support for including package-lock.json changes with lockfile option fixes #3

### LINKED ISSUE #3: Add option --lockfile to include package-lock.json content changes
Thanks for the improvements of install-changed package!
Would be nice to have a way to include package-lock.json file and package.json deps changes.

### PR DESCRIPTION
**Pull request by andidev**
Implemented support changes in package-lock.json through --lockfile option.

Tested both with and without flag.

[EDIT]
I added a --skip-write-hash flag to skip writing hash to file so it is possible to echo to console as warning in some cases when npm install is not run.

### CODE REVIEW

File: src/find-packagelock.ts
Comment (thdk) [inline src/find-packagelock.ts:19]: I think we can just reuse the existing `findPackage` function and pass an options object:
```
export function findPackage({
   useLockfile = false,
} = {}) {
   ...
   const search = path.join(current,  useLockfile ? 'package-lock.json' ; 'package.json');
   ...
}
```

File: src/is-package-changed.ts
Comment (thdk) [inline src/is-package-changed.ts:42]: No need to search for the lock file if the flag is off.

File: src/is-package-changed.ts
Comment (thdk) [inline src/is-package-changed.ts:50]: I would assume the `package-lock,json` file  contains all information needed and it would be no longer needed to get the hash from dependencies listed in `package.json` when `lockfile=true`.

File: src/is-package-changed.ts
Comment (thdk) [inline src/is-package-changed.ts:70]: (install) command has not been executed here yet, so the hash will not have been changed. I believe this code needs to run after line 71 and only for non CI environments.
```
let canWriteHash = await callback(result);
```

Comment (thdk) [reply src/is-package-changed.ts:70]: Also, on line 82 we expose a function that writes the hash on your file system. I think that we need to recalculate the hash there as well in case of `lockfile=true`

**Review by thdk — CHANGES_REQUESTED** [review]
I am not convinced yet about the use case for this, but since its behind an optional flag, I am willing to merge this when review comments have been addressed.

**Comment by thdk** [comment]
Can you revert the code related to `--skip-write-hash` to keep this PR related to a single feature?

File: src/is-package-changed.ts
Comment (thdk) [reply src/is-package-changed.ts:53]: Can you explain this?

File: src/find-packagelock.ts
Comment (thdk) [reply src/find-packagelock.ts]: Why thumbs down?

**Review by thdk — CHANGES_REQUESTED** [review]
Can you  give more information on why you reacted with 👎🏼  on some of the review comments?
Its important I understand what is going on and the code is not so recent in my head anymore.

Also this PR will not be merged if it contains more than one new feature. This to keep a clean history. And so we can discuss the need for that  flag separately from this flag.

**Comment by andidev** [comment]
> Can you give more information on why you reacted with 👎🏼 on some of the review comments? Its important I understand what is going on and the code is not so recent in my head anymore.
> 
> Also this PR will not be merged if it contains more than one new feature. This to keep a clean history. And so we can discuss the need for that flag separately from this flag.

Sorry maybe I should have not used that reactions. with thumbs down I meant I disagree and did not do the changes or solved it differently

**Comment by andidev** [comment]
> Can you revert the code related to `--skip-write-hash` to keep this PR related to a single feature?

well `--skip-write-hash` is really needed if using package-lock.json in the hash. Since otherwise it is not possible to detect a change and log some info and run plugin later with the npm run install.

I can separate them if really needed but it is pretty simple feature and does not affect PR that much. And if you decide on merging it as well I would rather not do that work since I really have not much time unfortunately.

**Comment by andidev** [comment]
See a really detailed comment for usecase:
https://github.com/thdk/package-changed/issues/3#issuecomment-997196973

**Comment by andidev** [comment]
Cant seem to answer below your comments for some reason.
Regarding
https://github.com/thdk/package-changed/pull/4#discussion_r771814600
> Can you explain this?
can you see my answers in the related issue?

**Comment by andidev** [comment]
@thdk so I removed the flag and will develop it in another PR once this is merged.

**Comment by thdk** [comment]
@andidev looks good now, I'll merge and release it asap.

One thing that I will still look at though if whether or not the package hash is still needed in case of using the lock file. As each change in package.json dependencies will always reflect in a different lock file.

Not blocking for no though.

(regarding to the `--skip-hash-file`, naming it `--no-hash-file` is already a plus :) )

**Comment by andidev** [comment]
@thdk 
Super thanks! I will  do the other change as well but regarding the name I am not sure you are right. 
First the name was skip-write-hash. However hash is still needed so --no-hash-file sound a bit like no hash file is used at all to check for change. But its up to you.

> @andidev looks good now, I'll merge and release it asap.
> 
> One thing that I will still look at though if whether or not the package hash is still needed in case of using the lock file. As each change in package.json dependencies will always reflect in a different lock file.

Well in case you modify package.json and dont run npm install that is still a change in dependencies. Its needed for my use case at least. See use case in related issue.

**Pull request merged**
